diff --git a/node_modules/postcss-css-variables/lib/is-under-scope.js b/node_modules/postcss-css-variables/lib/is-under-scope.js
index 7ce0baf..35e6a18 100644
--- a/node_modules/postcss-css-variables/lib/is-under-scope.js
+++ b/node_modules/postcss-css-variables/lib/is-under-scope.js
@@ -7,7 +7,7 @@ var RE_AT_RULE_SCOPE_PIECE  = (/^@.*/);
 // This will match pseudo selectors that have a base part
 // ex. .foo:hover
 // It will NOT match `:root`
-var RE_PSEUDO_SELECTOR = (/([^\s:]+)((?::|::)[^\s]*?)(\s+|$)/);
+var RE_PSEUDO_SELECTOR = (/([^\s:]+)(?<!\\)((?::|::)[^\s]*?)(\s+|$)/);
 
 
 function getScopeMatchResults(nodeScopeList, scopeNodeScopeList) {
diff --git a/node_modules/postcss-css-variables/lib/resolve-decl.js b/node_modules/postcss-css-variables/lib/resolve-decl.js
index e469849..5cfac55 100644
--- a/node_modules/postcss-css-variables/lib/resolve-decl.js
+++ b/node_modules/postcss-css-variables/lib/resolve-decl.js
@@ -137,12 +137,14 @@ function resolveDecl(decl, map, /*optional*/shouldPreserve, /*optional*/preserve
 				currentAtRuleNode = currentAtRuleNode.parent;
 			}
 
-			// Put the first atRuleStructure after the declaration's rule,
-			// and after that, put them right after the previous one
-			decl.parent.parent.insertAfter(preserveAtRulesOrder && previousAtRuleNode || decl.parent, parentAtRuleNode);
-
-			// Save referance of previous atRuleStructure
-			previousAtRuleNode = parentAtRuleNode
+			if (atRuleNode !== parentAtRuleNode) {
+				        // Put the first atRuleStructure after the declaration's rule,
+				        // and after that, put them right after the previous one
+				        decl.parent.parent.insertAfter(preserveAtRulesOrder && previousAtRuleNode || decl.parent, parentAtRuleNode);
+				
+				        // Save referance of previous atRuleStructure
+				        previousAtRuleNode = parentAtRuleNode
+				}
 		}
 		else {
 			ruleClone.selector = mimicDecl.parent.selector;
